api/v1alpha1/clusterextension_types.go

```diff
@@ -370,48 +361,45 @@ type ServiceAccountReference struct {
 	//
 	// [RFC 1123]: https://tools.ietf.org/html/rfc1123
 	//
-	//+kubebuilder:validation:MaxLength:=253
-	//+kubebuilder:validation:Pattern:=^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
-	//+kubebuilder:validation:XValidation:rule="self == oldSelf",message="name is immutable"
+	// +kubebuilder:validation:MaxLength:=253
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="name is immutable"
+	// +kubebuilder:validation:XValidation:rule="self.matches(\"^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$\")",message="name must be a valid DNS1123 subdomain"
+	// +kubebuilder:validation:Required
 	Name string `json:"name"`
 }
 
 // PreflightConfig holds the configuration for the preflight checks.  If used, at least one preflight check must be non-nil.
+//
 // +kubebuilder:validation:XValidation:rule="has(self.crdUpgradeSafety)",message="at least one of [crdUpgradeSafety] are required when preflight is specified"
 type PreflightConfig struct {
 	// crdUpgradeSafety is used to configure the CRD Upgrade Safety pre-flight
 	// checks that run prior to upgrades of installed content.
 	//
 	// The CRD Upgrade Safety pre-flight check safeguards from unintended
 	// consequences of upgrading a CRD, such as data loss.
-	//
-	// This field is required if the spec.install.preflight field is specified.
 	CRDUpgradeSafety *CRDUpgradeSafetyPreflightConfig `json:"crdUpgradeSafety"`
 }
 
 // CRDUpgradeSafetyPreflightConfig is the configuration for CRD upgrade safety preflight check.
 type CRDUpgradeSafetyPreflightConfig struct {
-	// policy is used to configure the state of the CRD Upgrade Safety pre-flight check.
-	//
-	// This field is required when the spec.install.preflight.crdUpgradeSafety field is
-	// specified.
+	// enforcement is a required field, used to configure the state of the CRD Upgrade Safety pre-flight check.
 	//
-	// Allowed values are ["Enabled", "Disabled"]. The default value is "Enabled".
+	// Allowed values are "None" or "Strict". The default value is "Strict".
 	//
-	// When set to "Disabled", the CRD Upgrade Safety pre-flight check will be skipped
+	// When set to "None", the CRD Upgrade Safety pre-flight check will be skipped
 	// when performing an upgrade operation. This should be used with caution as
 	// unintended consequences such as data loss can occur.
 	//
-	// When set to "Enabled", the CRD Upgrade Safety pre-flight check will be run when
+	// When set to "Strict", the CRD Upgrade Safety pre-flight check will be run when
 	// performing an upgrade operation.
 	//
-	//+kubebuilder:validation:Enum:="Enabled";"Disabled"
-	//+kubebuilder:default:=Enabled
-	Policy CRDUpgradeSafetyPolicy `json:"policy"`
+	// +kubebuilder:validation:Enum:="None";"Strict"
+	// +kubebuilder:default:=Strict
```

Comment:
I think @joelanford had originally called this out, but does this default actually do anything? We mark this as required but set a default value which I think tends to only ever be done on optional fields.

Reply:
We could view it as belt-and-suspenders, but practically I don't think it will have any effect. Removing.